### PR TITLE
feat: MVP share a query flow

### PR DESCRIPTION
**Pull request by niross**
### Description of change

- Allows sending an email to a data workspace user
- SQL is passed through in a query parameter
- Validates that the query is < 1950 chars - decided as ok for MVP

### Checklist

* [x] Have tests been added to cover any changes?

### UI Changes

**Step 1**
share-query-1

**Step 2**
share-query-2

**Step 3**
share-query-3

### CODE REVIEW

File: dataworkspace/dataworkspace/apps/explorer/forms.py [hunk 1]
```diff
@@ -56,3 +65,39 @@ def connections(self):
...
+    to_user = GOVUKDesignSystemEmailValidationModelChoiceField(
+        label='Who would you like to share the query with?',
+        help_text='Recipient\'s email',
+        queryset=get_user_model().objects.all(),
+        to_field_name='email',
+        widget=GOVUKDesignSystemTextWidget(label_is_heading=False),
+        required=True,
+        error_messages={
+            'invalid_email': 'Enter the email address in the correct format, for example name@digital.trade.gov.uk',
+            'invalid_choice': 'The user you are sharing with must have previously visited Data Workspace',
```
Comment (abbas123456) [inline dataworkspace/dataworkspace/apps/explorer/forms.py:80]: I think this error message may be a bit misleading, as the auth_user objects are synced from SSO, all SSO users will have a user object regardless of whether they have visited Data Workspace before

File: dataworkspace/dataworkspace/apps/explorer/templates/explorer/share_confirmation.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        <div class="govuk-panel govuk-panel--confirmation govuk-!-margin-bottom-6">
+          <h1 class="govuk-panel__title">Query shared</h1>
+        </div>
+        <h2 class="govuk-heading-m">What happens next</h2>
+        <p class="govuk-body">We’ve sent your query to {{ user.email }}</p>
+        <p class="govuk-body">
+          The person will receive your query and a direct link to the relavent page on Data Explorer along with your message.
+        </p>
+        <p class="govuk-body">
+          If the receiver does not have access to data, will have to request to be able to run it.
```
Comment (abbas123456) [inline dataworkspace/dataworkspace/apps/explorer/templates/explorer/share_confirmation.html:28]: Gramamatical errors

**Review by abbas123456 — APPROVED** [review]

File: dataworkspace/dataworkspace/apps/explorer/forms.py (hunk 1 above)
Comment (niross) [reply dataworkspace/dataworkspace/apps/explorer/forms.py:80]: True, will change it to `The user you are sharing with must have a DIT staff SSO account`

File: dataworkspace/dataworkspace/apps/explorer/templates/explorer/share_confirmation.html (hunk 2 above)
Comment (niross) [reply dataworkspace/dataworkspace/apps/explorer/templates/explorer/share_confirmation.html:28]: Ah yes good catch :+1:

**Pull request merged**